android/app/src/main/java/org/fossasia/openevent/dbutils/DatabaseOperations.java

```diff
@@ -625,6 +648,40 @@ public Speaker getSpeakerbySpeakersname(String speakerName, SQLiteDatabase mDb)
         return speaker;
     }
 
+    public Speaker getSpeakerId(int id, SQLiteDatabase mDb) {
+        String speakerColumnSelection = DbContract.Speakers.ID + EQUAL + id;
+        Cursor speakersCursor = mDb.query(
+                DbContract.Speakers.TABLE_NAME,
+                DbContract.Speakers.FULL_PROJECTION,
+                speakerColumnSelection,
+                null,
+                null,
+                null,
+                null
+        );
+        Speaker speaker = null;
+        speakersCursor.moveToFirst();
+            speaker = new Speaker(
+                    speakersCursor.getInt(speakersCursor.getColumnIndex(DbContract.Speakers.ID)),
+                    speakersCursor.getString(speakersCursor.getColumnIndex(DbContract.Speakers.NAME)),
+                    speakersCursor.getString(speakersCursor.getColumnIndex(DbContract.Speakers.PHOTO)),
+                    speakersCursor.getString(speakersCursor.getColumnIndex(DbContract.Speakers.BIO)),
+                    speakersCursor.getString(speakersCursor.getColumnIndex(DbContract.Speakers.EMAIL)),
+                    speakersCursor.getString(speakersCursor.getColumnIndex(DbContract.Speakers.WEB)),
+                    speakersCursor.getString(speakersCursor.getColumnIndex(DbContract.Speakers.TWITTER)),
+                    speakersCursor.getString(speakersCursor.getColumnIndex(DbContract.Speakers.FACEBOOK)),
+                    speakersCursor.getString(speakersCursor.getColumnIndex(DbContract.Speakers.GITHUB)),
+                    speakersCursor.getString(speakersCursor.getColumnIndex(DbContract.Speakers.LINKEDIN)),
+                    speakersCursor.getString(speakersCursor.getColumnIndex(DbContract.Speakers.ORGANISATION)),
+                    speakersCursor.getString(speakersCursor.getColumnIndex(DbContract.Speakers.POSITION)),
+                    null,
+                    speakersCursor.getString(speakersCursor.getColumnIndex(DbContract.Speakers.COUNTRY))
+
+            );
```

Comment:
This is a nifty way of getting the speaker if we have only the session ID with us.
I initially planned to display the speaker name too in the session card but dropped that idea later on.